packages/hash/api/src/model/util.ts

```diff
@@ -320,17 +319,17 @@ export const generateSystemEntityTypeSchema = (
     type: "object",
     kind: "entityType",
     properties,
-    required: requiredProperties.length > 0 ? requiredProperties : undefined,
+    required: requiredProperties,
     links,
-    requiredLinks: requiredLinks.length > 0 ? requiredLinks : undefined,
+    requiredLinks,
```

Comment:
I think this might make the tests fail, as we can end up with `requiredLinks: []` when there are no required links rather than `requiredLinks: undefined`. That means when we do an equals assertion between the created schema and the db schema the assertion could fail. Maybe I'm wrong about this though, just wanted to flag

Reply:
Should i run the tests to check? I made `outgoingLinks` optional, so it should bubble up here